lobby/src/test/java/org/triplea/lobby/server/login/LobbyLoginValidatorIntegrationTest.java

```diff
@@ -136,7 +118,9 @@ void testAnonymousLogin() {
     // create a user, verify we can't login with a username that already exists
     // we should not be able to login now
     assertNotNull(
-        generateChallenge(new HashedPassword(md5Crypt("foo"))).apply(challenge -> response));
+        generateChallenge(
+                new HashedPassword(BCrypt.hashpw(hashPasswordWithSalt("foo"), BCrypt.gensalt())))
+            .apply(challenge -> response));
```

Comment:
I could've sworn that there was a similar test case would exist somewhere in this file already (one for legacy, one for "new"), but it doesn't seem like that.

Reply:
Thanks for checking, having slightly redundant tests for now I think is okay:
- there are not too many redundant tests if any
- the extra coverage is not a bad trade-off for some potential redundancies, particularly for a section of code that is 'risky' (difficult to get absolutely correct and very important that it is correct)